shoes-swt/lib/shoes/swt/input_box.rb

```diff
@@ -20,6 +20,10 @@ def initialize(dsl, app, text_options)
         @real.add_modify_listener do |event|
           @dsl.call_change_listeners unless nothing_changed?(event)
         end
+
+        @real.add_verify_listener do |event|
+          event.doit = false if @readonly
```

Comment:
lol? :D There is an accessor in java called `doit` ? Surprises never cease, also nifty implementation!